src/compiler/scala/tools/nsc/ast/parser/Scanners.scala

```diff
@@ -40,6 +40,9 @@ trait ScannersCommon {
   }
 
   trait ScannerCommon extends CommonTokenData {
+    /** Consume and discard the next token. */
+    def nextToken(): Unit
```

Comment:
I had the feeling yesterday that the hierarchy is probably over-engineered or over-abstracted.